### PR TITLE
MAINT: version pins/prep for 1.14.0rc1

### PR DESCRIPTION
**Pull request by tylerjereddy**
* Draft in the version pins and related adjustments prior to the release of SciPy
`1.14.0rc1`.

[skip circle]

TODO:

- [x] Ralf checks version pins
- [x] Maybe check the wheel builds here to avoid surprises at release time proper

### CODE REVIEW

File: ci/cirrus_wheels.yml [hunk 1]
```diff
@@ -28,14 +28,7 @@ cirrus_wheels_linux_aarch64_task:
...
   env:
-    CIBW_PRERELEASE_PYTHONS: True
-    # TODO remove the CIBW_BEFORE_BUILD_LINUX line once there are numpy2.0 wheels available on PyPI.
-    # Also remove/comment out PIP_NO_BUILD_ISOLATION, PIP_EXTRA_INDEX_URL flags.
-    CIBW_BEFORE_BUILD_LINUX: "pip install numpy>=2.0.0.dev0 meson-python cython pythran pybind11 ninja;bash {project}/tools/wheels/cibw_before_build_linux.sh {project}"
-    CIBW_ENVIRONMENT: >
-      PIP_PRE=1
-      PIP_EXTRA_INDEX_URL=https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
-      PIP_NO_BUILD_ISOLATION=false
+    CIBW_PRERELEASE_PYTHONS: False
```
Comment (tylerjereddy) [inline ci/cirrus_wheels.yml:31]: could `main` be adjusted so I don't need to repeat this or no?

Comment (rgommers) [reply ci/cirrus_wheels.yml:31]: Yes, I think so. This same thing was already removed from `.github/workflows/wheels.yml` in gh-20585.

**Review by rgommers — APPROVED** [review]
Changes all LGTM, thanks Tyler. Feel free to merge or to trigger wheel builds - whatever you prefer.

**Comment by tylerjereddy** [comment]
Wheel builds seem "ok," I had to restart one for a connection issue.

**Pull request merged**